src/plugins/validation/2and3/semantic-validators/pagination-ibm.js

```diff
@@ -0,0 +1,181 @@
+module.exports.validate = function({ jsSpec }, config) {
+  const result = {};
+  result.error = [];
+  result.warning = [];
+
+  config = config.pagination;
+  const queryParameters = [];
+  for (const head in jsSpec.paths) {
+    if (/{/.test(head)) {
+      continue;
+    }
+    if (jsSpec.paths[head].get) {
+      for (const resp in jsSpec.paths[head].get.responses) {
+        for (const con in jsSpec.paths[head].get.responses[resp].content) {
+          if (con == 'application/json') {
+            let arrayOnTop = false;
+            if (
+              (jsSpec.paths[head].get.responses[resp].content[con].schema &&
+                jsSpec.paths[head].get.responses[resp].content[con].schema
+                  .properties) ||
+              (jsSpec.paths[head].get.responses[resp].content[con].schema
+                .pagination &&
+                jsSpec.paths[head].get.responses[resp].content[con].schema
+                  .pagination.properties)
+            ) {
+              for (const prop in jsSpec.paths[head].get.responses[resp].content[
+                con
+              ].schema.properties) {
+                if (
+                  jsSpec.paths[head].get.responses[resp].content[con].schema
+                    .properties[prop].type == 'array'
+                ) {
+                  arrayOnTop = true;
+                  break;
+                }
+              }
+              for (
+                let k = 0;
+                k < jsSpec.paths[head].get.parameters.length;
+                k++
+              ) {
+                queryParameters.push(jsSpec.paths[head].get.parameters[k].name);
+              }
```

Comment:
If all you want is a list of the parameter names, there is a much more elegant and concise way to do that using the `map` operator.

A good article on this: https://medium.com/poka-techblog/simplify-your-javascript-use-map-reduce-and-filter-bd02c593cc2d